samples/auth/src/main/java/org/forgerock/auth/MainActivity.java

```diff
@@ -97,33 +114,14 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
             // Make sure the request was successful
             if (resultCode == RESULT_OK) {
                 success.setVisibility(VISIBLE);
-                if (FRUser.getCurrentUser() != null) {
-                    FRUser.getCurrentUser().getUserInfo(new FRListener<UserInfo>() {
-                        @Override
-                        public void onSuccess(final UserInfo result) {
-                            runOnUiThread(() -> {
-                                progressBar.setVisibility(INVISIBLE);
-                                try {
-                                    content.setText(result.getRaw().toString(2));
-                                } catch (JSONException e) {
-                                    onException(e);
-                                }
-                            });
-                        }
-
-                        @Override
-                        public void onException(final Exception e) {
-                            runOnUiThread(() -> {
-                                progressBar.setVisibility(INVISIBLE);
-                                content.setText(e.getMessage());
-                            });
-                        }
-                    });
-                } else {
-                    Snackbar.make(findViewById(org.forgerock.auth.R.id.success), "No Session", LENGTH_LONG).show();
-                }
+                userinfo();
             } else {
-                Snackbar.make(findViewById(org.forgerock.auth.R.id.success), "Login Failed", LENGTH_LONG).show();
+                Exception exception = null;
+                if (data != null) {
+                    exception = (Exception) data.getSerializableExtra(SimpleLoginActivity.EXCEPTION);
+                }
+                Snackbar.make(findViewById(org.forgerock.auth.R.id.success), "Login Failed:" +
```

Comment:
Is Snackbar a UI related functionality? I ask because this portion of code is considered Core (non-UI), yeah?